whelk-core/src/main/groovy/whelk/JsonLd.groovy

```diff
@@ -506,6 +505,50 @@ class JsonLd {
         return true
     }
 
+    static List<List<String>> findPaths(Map obj, String key, String value) {
+        List paths = []
+        new DFS().search(obj, { List path, v ->
+            if (value == v && key == path[-1]) {
+                paths << new ArrayList(path)
+            }
+        })
+        return paths
+    }
+
+    private static class DFS {
+        interface Callback {
+            void node(List path, value)
+        }
+
+        List path = []
+        Callback cb
+
+        void search(obj, Callback callback) {
+            cb = callback
+            path = []
+            node(obj)
+        }
+
+        private void node(obj) {
+            cb.node(path, obj)
+            if (obj instanceof Map) {
+                descend(((Map) obj).entrySet().collect({ new Tuple2(it.value, it.key) }))
+            } else if (obj instanceof List) {
+                descend(((List) obj).withIndex())
+            }
+        }
+
+        private void descend(List<Tuple2> nodes) {
+            for (n in nodes) {
+                path << n.second
+                node(n.first)
+                path.remove(path.size()-1)
```

Comment:
List.pop() removes the first item in Groovy 2.5 (!) and removeLast() is not available before 2.5.